service/dap/server.go

```diff
@@ -550,58 +550,66 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		s.config.Debugger.WorkingDir = wdParsed
 	}
 
-	noDebug := false
-	if noDebugArg, ok := request.Arguments["noDebug"]; ok {
-		if v, ok := noDebugArg.(bool); ok && v { // noDebug == true
-			noDebug = true
-		}
-	}
-
-	var err error
-	if noDebug {
-		if s.noDebugProcess, err = s.launchWithoutDebug(program, targetArgs, s.config.Debugger.WorkingDir); err != nil {
+	if noDebug, _ := request.Arguments["noDebug"].(bool); noDebug {
+		cmd, err := s.startNoDebugProcess(program, targetArgs, s.config.Debugger.WorkingDir)
+		if err != nil {
 			s.sendErrorResponse(request.Request, FailedToLaunch, "Failed to launch", err.Error())
 			return
 		}
-	} else {
-		if s.debugger, err = debugger.New(&s.config.Debugger, s.config.ProcessArgs); err != nil {
-			s.sendErrorResponse(request.Request, FailedToLaunch, "Failed to launch", err.Error())
-			return
+		// Skip 'initialized' event.
+		s.send(&dap.LaunchResponse{Response: *newResponse(request.Request)})
+
+		// Then, block until the program terminates or is stopped.
+		if err := cmd.Wait(); err != nil {
+			s.log.Errorf("program exited: %v", err)
 		}
+
+		stopped := false
+		s.mu.Lock()
+		stopped = s.noDebugProcess == nil // if it was stopped, this should be nil.
+		s.noDebugProcess = nil
+		s.mu.Unlock()
+
+		if !stopped {
+			s.send(&dap.TerminatedEvent{Event: *newEvent("terminated")})
+		}
+		return
+	}
+
+	var err error
+	if s.debugger, err = debugger.New(&s.config.Debugger, s.config.ProcessArgs); err != nil {
```

Comment:
does err := not work? (to avoid "var err error")

Reply:
`:=` only works with variable names on the left.

Reply:
Ah, I see, "non-name ... on left side of :=". That's unfortunate.

Reply:
The offical name of := is a short variable declaration, and 's.debugger' isn't a variable.